### PR TITLE
[OCPCLOUD-1584] Add webhook validations for Azure boot diagnostics

### PR DESCRIPTION
**Pull request by JoelSpeed**
This PR updates the API dependency to include the new Azure boot diagnostics configuration and then sets up webhook validations to prevent invalid user configuration

### CODE REVIEW

File: pkg/webhooks/machine_webhook.go [hunk 1]
```diff
@@ -1027,6 +1029,30 @@ func validateAzureImage(image machinev1beta1.Image) []error {
 	return errors
 }
 
+func validateAzureDiagnostics(spec machinev1beta1.AzureDiagnostics, parentPath *field.Path) []error {
```
Comment (lobziik) [inline pkg/webhooks/machine_webhook.go:1032]: nit: suggest to rename `spec` to smth like `diagnosticSpec`, just for make this name more descriptive within function.

**Review by damdo — COMMENTED** [review]
Provisionally LGTM.
We'll need to update go.mod once the branch reopens and the api changes land.

**Review by elmiko — COMMENTED** [review]
makes sense to me
/lgtm

**Review by alexander-demichev — APPROVED** [review]
/approve

**Pull request merged**